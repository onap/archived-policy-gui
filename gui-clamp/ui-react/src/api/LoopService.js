/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

export default class LoopService {
	static getLoopNames() {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/getAllNames', { method: 'GET', credentials: 'same-origin' })
			.then(function (response) {
				console.debug("GetLoopNames response received: ", response.status);
				if (response.ok) {
					return response.json();
				} else {
					console.error("GetLoopNames query failed");
					return {};
				}
			})
			.catch(function (error) {
				console.error("GetLoopNames error received", error);
				return {};
			});
	}

	static createLoop(loopName, templateName) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/create/' + loopName + '?templateName=' + templateName, {
			method: 'POST',
			headers: {
				"Content-Type": "application/json"
			},
			credentials: 'same-origin'
		})
			.then(function (response) {
				console.debug("CreateLoop response received: ", response.status);
				return response.json();
			})
			.catch(function (error) {
				console.error("CreateLoop error received", error);
				return "";
			});
	}

	static getLoop(loopName) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/' + loopName, {
			method: 'GET',
			headers: {
				"Content-Type": "application/json"
			},
			credentials: 'same-origin'
		})
			.then(function (response) {
				console.debug("GetLoop response received: ", response.status);
				if (response.ok) {
					return response.json();
				} else {
					console.error("GetLoop query failed");
					return {};
				}
			})
			.catch(function (error) {
				console.error("GetLoop error received", error);
				return {};
			});
	}

	static setMicroServiceProperties(loopName, jsonData) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/updateMicroservicePolicy/' + loopName, {
			method: 'POST',
			credentials: 'same-origin',
			headers: {
				"Content-Type": "application/json"
			},
			body: JSON.stringify(jsonData)
		})
			.then(function (response) {
				console.debug("updateMicroservicePolicy response received: ", response.status);
				if (response.ok) {
					return response.text();
				} else {
					console.error("updateMicroservicePolicy query failed");
					return "";
				}
			})
			.catch(function (error) {
				console.error("updateMicroservicePolicy error received", error);
				return "";
			});
	}
	
	static setOperationalPolicyProperties(loopName, jsonData) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/updateOperationalPolicies/' + loopName, {
			method: 'POST',
			credentials: 'same-origin',
			headers: {
				"Content-Type": "application/json"
			},
			body: JSON.stringify(jsonData)
		})
			.then(function (response) {
				console.debug("updateOperationalPolicies response received: ", response.status);
				if (response.ok) {
					return response.text();
				} else {
					console.error("updateOperationalPolicies query failed");
					return "";
				}
			})
			.catch(function (error) {
				console.error("updateOperationalPolicies error received", error);
				return "";
			});
	}

	static updateGlobalProperties(loopName, jsonData) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/updateGlobalProperties/' + loopName, {
			method: 'POST',
			credentials: 'same-origin',
			headers: {
				"Content-Type": "application/json"
			},
			body: JSON.stringify(jsonData)
		})
			.then(function (response) {
				console.debug("updateGlobalProperties response received: ", response.status);
				if (response.ok) {
					return response.text();
				} else {
					console.error("updateGlobalProperties query failed");
					return "";
				}
			})
			.catch(function (error) {
				console.error("updateGlobalProperties error received", error);
				return "";
			});
	}

	static refreshOperationalPolicyJson(loopName,operationalPolicyName) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/refreshOperationalPolicyJsonSchema/' + loopName + '/' + operationalPolicyName, {
			method: 'PUT',
			headers: {
				"Content-Type": "application/json"
			},
			credentials: 'same-origin'
		})
			.then(function (response) {
				console.debug("Refresh Operational Policy Json Schema response received: ", response.status);
				if (response.ok) {
					return response.json();
				} else {
					console.error("Refresh Operational Policy Json Schema query failed");
					return {};
				}
			})
			.catch(function (error) {
				console.error("Refresh Operational Policy Json Schema error received", error);
				return {};
			});
	}

		static refreshMicroServicePolicyJson(loopName,microServicePolicyName) {
    		return fetch(window.location.pathname + 'restservices/clds/v2/loop/refreshMicroServicePolicyJsonSchema/' + loopName + '/' + microServicePolicyName, {
    			method: 'PUT',
    			headers: {
    				"Content-Type": "application/json"
    			},
    			credentials: 'same-origin'
    		})
    			.then(function (response) {
    				console.debug("Refresh Operational Policy Json Schema response received: ", response.status);
    				if (response.ok) {
    					return response.json();
    				} else {
    					console.error("Refresh Operational Policy Json Schema query failed");
    					return {};
    				}
    			})
    			.catch(function (error) {
    				console.error("Refresh Operational Policy Json Schema error received", error);
    				return {};
    			});
    }

	static addOperationalPolicyType(loopName, policyType, policyVersion) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/addOperationaPolicy/' + loopName + '/policyModel/' + policyType +'/' + policyVersion , {
			method: 'PUT',
			headers: {
				"Content-Type": "application/json"
			},
			credentials: 'same-origin'
		})
				.then(function (response) {
				console.debug("Add Operational Policy response received: ", response.status);
				if (response.ok) {
					return response.json();
				} else {
					return response.text();
				}
			})
			.catch(function (error) {
				console.error("Add Operational Policy query failed");
				throw new Error(error);
			})
	}

	static removeOperationalPolicyType(loopName, policyType, policyVersion, policyName) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/removeOperationaPolicy/' + loopName + '/policyModel/' + policyType +'/' + policyVersion + '/' + policyName , {
			method: 'PUT',
			headers: {
				"Content-Type": "application/json"
			},
			credentials: 'same-origin'
		})
				.then(function (response) {
					console.debug("Remove Operational Policy response received: ", response.status);
				if (response.ok) {
					return response.json();
				} else {
					console.error("Remove Operational Policy query failed");
					return {};
				}
			})
			.catch(function (error) {
				console.error("Remove Operational Policy error received", error);
				return {};
			});
	}
}
