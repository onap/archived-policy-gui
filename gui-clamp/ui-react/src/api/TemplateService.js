/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

export default class TemplateService {

  static getLoopNames() {
    return fetch(window.location.pathname + 'restservices/clds/v2/loop/getAllNames', { method: 'GET', credentials: 'same-origin' })
      .then(function (response) {
        console.debug("getLoopNames response received: ", response.status);
        if (response.ok) {
          return response.json();
        } else {
          console.error("getLoopNames query failed");
          return {};
        }
      })
      .catch(function (error) {
        console.error("getLoopNames error received", error);
        return {};
      });
  }

  static getAllLoopTemplates() {
    return fetch(window.location.pathname + 'restservices/clds/v2/templates', { method: 'GET', credentials: 'same-origin', })
      .then(function (response) {
        console.debug("getAllLoopTemplates response received: ", response.status);
        if (response.ok) {
          return response.json();
        } else {
          console.error("getAllLoopTemplates query failed");
          return {};
        }
      })
      .catch(function (error) {
        console.error("getAllLoopTemplates error received", error);
        return {};
      });
  }

  static getDictionary() {
    return fetch(window.location.pathname + 'restservices/clds/v2/dictionary/', { method: 'GET', credentials: 'same-origin', })
      .then(function (response) {
        console.debug("getDictionary response received: ", response.status);
        if (response.ok) {
          return response.json();
        } else {
          console.error("getDictionary query failed");
          return {};
        }
      })
      .catch(function (error) {
        console.error("getDictionary error received", error);
        return {};
      });
  }

  static getDictionaryElements(dictionaryName) {
    return fetch(window.location.pathname + 'restservices/clds/v2/dictionary/' + dictionaryName, {
      method: 'GET',
      headers: {
        "Content-Type": "application/json",
      },
      credentials: 'same-origin',
    })
      .then(function (response) {
        console.debug("getDictionaryElements response received: ", response.status);
        if (response.ok) {
          return response.json();
        } else {
          console.error("getDictionaryElements query failed");
          return {};
        }
      })
      .catch(function (error) {
        console.error("getDictionaryElements error received", error);
        return {};
      });
  }

  static insDictionary(jsonData) {
    console.log("dictionaryName is", jsonData.name)
    return fetch(window.location.pathname + 'restservices/clds/v2/dictionary/', {
      method: 'PUT',
      credentials: 'same-origin',
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(jsonData)
    })
      .then(function (response) {
        console.debug("insDictionary response received: ", response.status);
        if (response.ok) {
          return response.status;
        } else {
          var errorMessage = response.status;
          console.error("insDictionary query failed", response.status);
          return errorMessage;
        }
      })
      .catch(function (error) {
        console.error("insDictionary error received", error);
        return "";
      });
  }

  static insDictionaryElements(jsonData) {
    console.log("dictionaryName is", jsonData.name)
    return fetch(window.location.pathname + 'restservices/clds/v2/dictionary/' + jsonData.name, {
      method: 'PUT',
      credentials: 'same-origin',
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(jsonData)
    })
      .then(function (response) {
        console.debug("insDictionary response received: ", response.status);
        if (response.ok) {
          return response.status;
        } else {
          var errorMessage = response.status;
          console.error("insDictionary query failed", response.status);
          return errorMessage;
        }
      })
      .catch(function (error) {
        console.error("insDictionary error received", error);
        return "";
      });
  }

  static deleteDictionary(dictionaryName) {
    console.log("inside templaemenu service", dictionaryName)
    return fetch(window.location.pathname + 'restservices/clds/v2/dictionary/' + dictionaryName, {
      method: 'DELETE',
      headers: {
        "Content-Type": "application/json",
      },
      credentials: 'same-origin',
    })
      .then(function (response) {
        console.debug("deleteDictionary response received: ", response.status);
        if (response.ok) {
          return response.status;
        } else {
          console.error("deleteDictionary query failed");
          return {};
        }
      })
      .catch(function (error) {
        console.error("deleteDictionary error received", error);
        return {};
      });
  }

  static deleteDictionaryElements(dictionaryData) {
    return fetch(window.location.pathname + 'restservices/clds/v2/dictionary/' + dictionaryData.name + '/elements/' + dictionaryData.shortName, {
      method: 'DELETE',
      headers: {
        "Content-Type": "application/json",
      },
      credentials: 'same-origin',
    })
      .then(function (response) {
        console.debug("deleteDictionary response received: ", response.status);
        if (response.ok) {
          return response.status;
        } else {
          console.error("deleteDictionary query failed");
          return {};
        }
      })
      .catch(function (error) {
        console.error("deleteDictionary error received", error);
        return {};
      });
  }
}
