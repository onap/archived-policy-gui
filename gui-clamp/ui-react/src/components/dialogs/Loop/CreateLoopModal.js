/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2020 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

import React from 'react'
import Select from 'react-select';
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';
import Form from 'react-bootstrap/Form';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import styled from 'styled-components';
import LoopService from '../../../api/LoopService';
import TemplateService from '../../../api/TemplateService';
import LoopCache from '../../../api/LoopCache';
import SvgGenerator from '../../loop_viewer/svg/SvgGenerator';

const ModalStyled = styled(Modal)`
	background-color: transparent;
`

const ErrMsgStyled = styled.div`
	color: red;
`

export default class CreateLoopModal extends React.Component {
	constructor(props, context) {
		super(props, context);

		this.getAllLoopTemplates = this.getAllLoopTemplates.bind(this);
		this.handleCreate = this.handleCreate.bind(this);
		this.handleModelName = this.handleModelName.bind(this);
		this.handleClose = this.handleClose.bind(this);
		this.handleDropDownListChange = this.handleDropDownListChange.bind(this);
		this.renderSvg = this.renderSvg.bind(this);
		this.state = {
			show: true,
			chosenTemplateName: '',
			modelInputErrMsg: '',
			modelName: '',
			templateNames: [],
			fakeLoopCacheWithTemplate: new LoopCache({})
		};
	}

	async componentDidMount() {
		await this.getAllLoopTemplates();
		await this.getModelNames();
	}

	handleClose() {
		this.setState({ show: false });
		this.props.history.push('/');
	}

	handleDropDownListChange(e) {
	    if (typeof e.value !== "undefined") {
	        this.setState({
	        fakeLoopCacheWithTemplate:
                new LoopCache({
                    "loopTemplate":e.templateObject,
                    "name": "fakeLoop"
                }),
                chosenTemplateName: e.value
	        })
		} else {
          	this.setState({ fakeLoopCacheWithTemplate: new LoopCache({}) })
        }
	}

	getAllLoopTemplates() {
		TemplateService.getAllLoopTemplates().then(templatesData => {
		    const templateOptions = templatesData.map((templateData) => { return { label: templateData.name, value: templateData.name, templateObject: templateData } });
            this.setState({
                templateNames: templateOptions })
		});
	}

	getModelNames() {
		TemplateService.getLoopNames().then(loopNames => {
			if (!loopNames) {
				loopNames = [];
			}
			// Remove LOOP_ prefix
			let trimmedLoopNames = loopNames.map(str => str.replace('LOOP_', ''));
			this.setState({ modelNames: trimmedLoopNames });
		});
	}

	handleCreate() {
		if (!this.state.modelName) {
			alert("A model name is required");
			return;
		}
		console.debug("Create Model " + this.state.modelName + ", Template " + this.state.chosenTemplateName + " is chosen");
		this.setState({ show: false });
		LoopService.createLoop("LOOP_" + this.state.modelName, this.state.chosenTemplateName).then(text => {
			console.debug("CreateLoop response received: ", text);
			try {
				this.props.history.push('/');
				this.props.loadLoopFunction("LOOP_" + this.state.modelName);
			} catch(err) {
				alert(text);
				this.props.history.push('/');
			}
		})
		.catch(error => {
			console.debug("Create Loop failed");
		});
	}

	handleModelName(event) {
		if (this.state.modelNames.includes(event.target.value)) {
			this.setState({
				modelInputErrMsg: 'A model named "' + event.target.value + '" already exists. Please pick another name.',
				modelName: event.target.value
			});
			return;
		} else {
			this.setState({
				modelInputErrMsg: '',
				modelName: event.target.value
			});
		}
	}

	renderSvg() {
		return (
			<SvgGenerator loopCache={this.state.fakeLoopCacheWithTemplate} clickable={false} generatedFrom={SvgGenerator.GENERATED_FROM_TEMPLATE}/>
		);
	}

	render() {
		return (
			<ModalStyled size="xl" show={this.state.show} onHide={this.handleClose} backdrop="static" keyboard={false} >
				<Modal.Header closeButton>
					<Modal.Title>Create Model</Modal.Title>
				</Modal.Header>
				<Modal.Body>
					<Form.Group as={Row} controlId="formPlaintextEmail">
						<Form.Label column sm="2">Template Name:</Form.Label>
						<Col sm="10">
							<Select onChange={this.handleDropDownListChange} options={this.state.templateNames} />
						</Col>
					</Form.Group>
                    <Form.Group as={Row} style={{alignItems: 'center'}} controlId="formSvgPreview">
                    <Form.Label column sm="2">Model Preview:</Form.Label>
                        <Col sm="10">
                            {this.renderSvg()}
                        </Col>
                    </Form.Group>
					<Form.Group as={Row} controlId="formPlaintextEmail">
						<Form.Label column sm="2">Model Name:</Form.Label>
						<input sm="5" type="text" style={{width: '50%', marginLeft: '1em' }}
							value={this.state.modelName}
							onChange={this.handleModelName}
						/>
						<span sm="5"/>
					</Form.Group>
					<Form.Group as={Row} controlId="formPlaintextEmail">
						<Form.Label column sm="2"> </Form.Label>
						<ErrMsgStyled>{this.state.modelInputErrMsg}</ErrMsgStyled>
					</Form.Group>
				</Modal.Body>
				<Modal.Footer>
					<Button variant="secondary" type="null" onClick={this.handleClose}>Cancel</Button>
					<Button variant="primary" type="submit" onClick={this.handleCreate}>Create</Button>
				</Modal.Footer>
			</ModalStyled>
		);
	}
}
