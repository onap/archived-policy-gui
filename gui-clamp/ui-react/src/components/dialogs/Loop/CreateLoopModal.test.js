/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2020 AT&T Intellectual Property. All rights
 *                             reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */
import React from 'react';
import { shallow } from 'enzyme';
import CreateLoopModal from './CreateLoopModal';
import LoopService from '../../../api/LoopService';
import TemplateService from '../../../api/TemplateService';

describe('Verify CreateLoopModal', () => {

  it('Test the render method', async () => {
    const flushPromises = () => new Promise(setImmediate);
    TemplateService.getAllLoopTemplates = jest.fn().mockImplementation(() => {
      return Promise.resolve([{"name":"template1"},{"name":"template2"}]);
    });
    TemplateService.getLoopNames = jest.fn().mockImplementation(() => {
      return Promise.resolve([]);
    });

    const component = shallow(<CreateLoopModal/>);
    expect(component).toMatchSnapshot();
    await flushPromises();
    component.update();
    expect(component.state('templateNames')).toStrictEqual([{"label": "template1", "value": "template1", "templateObject": {"name": "template1"}}, {"label": "template2", "value": "template2","templateObject": {"name": "template2"}}]);
  });

  it('handleDropdownListChange event', async () => {
  	const flushPromises = () => new Promise(setImmediate);

    const component = shallow(<CreateLoopModal/>);
    component.find('StateManager').simulate('change', {value: 'template1', templateObject: {"name":"template1"} });
    await flushPromises();
    component.update();
    expect(component.state('chosenTemplateName')).toEqual("template1");
    expect(component.state('fakeLoopCacheWithTemplate').getLoopTemplate()['name']).toEqual("template1");
    expect(component.state('fakeLoopCacheWithTemplate').getLoopName()).toEqual("fakeLoop");

	component.find('StateManager').simulate('change',{value: 'template2', templateObject: {"name":"template2"} });
    await flushPromises();
    component.update();
    expect(component.state('chosenTemplateName')).toEqual("template2");
    expect(component.state('fakeLoopCacheWithTemplate').getLoopTemplate()['name']).toEqual("template2");
    expect(component.state('fakeLoopCacheWithTemplate').getLoopName()).toEqual("fakeLoop");
  });

  it('handleModelName event', async () => {
    const flushPromises = () => new Promise(setImmediate);
    TemplateService.getAllLoopTemplates = jest.fn().mockImplementation(() => {
      return Promise.resolve([{"name":"template1"},{"name":"template2"}]);
    });
    TemplateService.getLoopNames = jest.fn().mockImplementation(() => {
      return Promise.resolve([]);
    });
    const event = {target: {value : "model1"} };
    const component = shallow(<CreateLoopModal/>);
    await flushPromises();
    component.find('input').simulate('change', event);
    component.update();
    expect(component.state('modelName')).toEqual("model1");
  });

  it('Test handleClose', () => {
    const historyMock = { push: jest.fn() }; 
    const handleClose = jest.spyOn(CreateLoopModal.prototype,'handleClose');
    const component = shallow(<CreateLoopModal history={historyMock} />)

    component.find('[variant="secondary"]').prop('onClick')();

    expect(handleClose).toHaveBeenCalledTimes(1);
    expect(component.state('show')).toEqual(false);
    expect(historyMock.push.mock.calls[0]).toEqual([ '/']);
    
    handleClose.mockClear();
  });

  it('Test handleCreate Fail', () => {
    const handleCreate = jest.spyOn(CreateLoopModal.prototype,'handleCreate');
    const component = shallow(<CreateLoopModal/>)

    component.find('[variant="primary"]').prop('onClick')();

    expect(handleCreate).toHaveBeenCalledTimes(1);
    expect(component.state('show')).toEqual(true);

    handleCreate.mockClear();
  });

  it('Test handleCreate Suc', async () => {
	const flushPromises = () => new Promise(setImmediate);
    const historyMock = { push: jest.fn() };
    const loadLoopFunction = jest.fn();  
    
    LoopService.createLoop = jest.fn().mockImplementation(() => {
			return Promise.resolve({
				ok: true,
				status: 200,
				json: () => {}
			});
		});

    const handleCreate = jest.spyOn(CreateLoopModal.prototype,'handleCreate');
    const component = shallow(<CreateLoopModal history={historyMock} loadLoopFunction={loadLoopFunction}/>)
    component.setState({
			modelName: "modelNameTest",
			chosenTemplateName: "template1"
		});

    component.find('[variant="primary"]').prop('onClick')();
	await flushPromises();
	component.update();

    expect(handleCreate).toHaveBeenCalledTimes(1);
    expect(component.state('show')).toEqual(false);
    expect(historyMock.push.mock.calls[0]).toEqual([ '/']);

    handleCreate.mockClear();
  });

});
