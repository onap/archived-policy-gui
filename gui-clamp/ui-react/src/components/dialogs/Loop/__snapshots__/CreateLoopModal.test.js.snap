// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify CreateLoopModal Test the render method 1`] = `
<Styled(Modal)
  backdrop="static"
  keyboard={false}
  onHide={[Function]}
  show={true}
  size="xl"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Create Model
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "noGutters": false,
          },
          "render": [Function],
        }
      }
      controlId="formPlaintextEmail"
    >
      <FormLabel
        column={true}
        sm="2"
        srOnly={false}
      >
        Template Name:
      </FormLabel>
      <Col
        sm="10"
      >
        <StateManager
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          onChange={[Function]}
          options={Array []}
        />
      </Col>
    </FormGroup>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "noGutters": false,
          },
          "render": [Function],
        }
      }
      controlId="formSvgPreview"
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <FormLabel
        column={true}
        sm="2"
        srOnly={false}
      >
        Model Preview:
      </FormLabel>
      <Col
        sm="10"
      >
        <withRouter(SvgGenerator)
          clickable={false}
          generatedFrom="TEMPLATE"
          loopCache={
            LoopCache {
              "loopJsonCache": Object {},
            }
          }
        />
      </Col>
    </FormGroup>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "noGutters": false,
          },
          "render": [Function],
        }
      }
      controlId="formPlaintextEmail"
    >
      <FormLabel
        column={true}
        sm="2"
        srOnly={false}
      >
        Model Name:
      </FormLabel>
      <input
        onChange={[Function]}
        sm="5"
        style={
          Object {
            "marginLeft": "1em",
            "width": "50%",
          }
        }
        type="text"
        value=""
      />
      <span
        sm="5"
      />
    </FormGroup>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "noGutters": false,
          },
          "render": [Function],
        }
      }
      controlId="formPlaintextEmail"
    >
      <FormLabel
        column={true}
        sm="2"
        srOnly={false}
      >
         
      </FormLabel>
      <styled.div />
    </FormGroup>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="null"
      variant="secondary"
    >
      Cancel
    </Button>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="submit"
      variant="primary"
    >
      Create
    </Button>
  </ModalFooter>
</Styled(Modal)>
`;
